### DISCUSSION TITLE
@export should imply @tool

### DISCUSSION THREAD

**Discussion by servel333** [godotengine/godot-proposals#8417] [Scripting]
I would like to explore with the community the idea that `@tool` is not required if the script contains any `@export` and `@export_*`s annotations.

- Does the `@export` have utility when the script is not a `@tool`?
- Is there some planned, or possible utility to that annotation?
- Thoughts?

The proposed change here is to make it so that if any script contains any `@export` or `@export_*`s, then the `@tool` is implied and the script will be treated as though it has `@tool` in it.

I searched `"@export" should imply "@tool"` and found nothing.

**Comment by AThousandShips** [comment]
`@tool` is not required for `@export` itself, and it is not implied, see [here](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_exports.html)

**Reply by Calinou** [reply]
Indeed, I can confirm there are plenty of cases where I want to have exported properties but don't want the script to ever run in the editor. The most common use case is when you have a character and want to be able to edit their statistics (movement speed, health, …) without having to edit the script.

**Comment by theraot** [comment]
> I would like to explore with the community the idea that @tool is not required if the script contains any @export and @export_*s annotations.

I have scripts that use `@tool` and don't have any `@export`, usually because they are intended to be called from other scripts that have `@tool`, and sometimes because they are part of an addon with an `EditorPlugin`.

> Does the `@export` have utility when the script is not a `@tool`?

Yes. Godot will save whatever you set in the inspector (be that a variable with `@export` or not) and set it when the game runs (barring property usage flags when using `_get_property_list`).

> Is there some planned, or possible utility to that annotation?

If there was, there would be a proposal up. I don't recall seeing any.

> Thoughts?

Why do you think `@export` should imply `@tool`?

Please notice that `@tool` would also make other things run in the editor aside from setters including `_ready`, `_process` and `_physics_process`. So you might be overcorrecting for something... I don't know what, so what is the issue you are trying to correct for?

**Reply by AThousandShips** [reply]
Note additionally that the `_get_property_list` method is still being discussed AFAIK, and whether or not to allow running it without `@tool`, as well as allowing you to annotate individual methods as `@tool` to allow the same, as `@tool` is a bit "all or nothing", see https://github.com/godotengine/godot-proposals/issues/729

**Comment by servel333** [comment]
I didn't know these extra uses for those annotations so I'm convinced that this proposal would erase some useful functionality.
